src/app/WriteClient.cpp

```diff
@@ -449,30 +461,40 @@ CHIP_ERROR WriteClient::OnMessageReceived(Messaging::ExchangeContext * apExchang
     }
     else if (aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::StatusResponse))
     {
-        err = StatusResponse::ProcessStatusResponse(std::move(aPayload));
-        SuccessOrExit(err);
+        CHIP_ERROR statusError = CHIP_NO_ERROR;
+        SuccessOrExit(err = StatusResponse::ProcessStatusResponse(std::move(aPayload), statusError));
+        SuccessOrExit(err = statusError);
+        err = CHIP_ERROR_INVALID_MESSAGE_TYPE;
     }
     else
     {
         err = CHIP_ERROR_INVALID_MESSAGE_TYPE;
     }
 
 exit:
-    if (mpCallback != nullptr)
+    ResponseMessageHandled(err, apExchangeContext, suppressErrorStatusResponse);
+    return err;
+}
+
+void WriteClient::ResponseMessageHandled(CHIP_ERROR aError, Messaging::ExchangeContext * apExchangeContext,
+                                         bool aSuppressErrorStatusResponse)
+{
+    if (aError != CHIP_NO_ERROR)
     {
-        if (err != CHIP_NO_ERROR)
+        StatusResponse::Send(Protocols::InteractionModel::Status::InvalidAction, apExchangeContext, false /*aExpectResponse*/);
+        if (!aSuppressErrorStatusResponse)
```

Comment:
This doesn't make any sense.  Why is `aSuppressErrorStatusResponse` affecting calls to `mpCallback` but not affecting the sending of the status response?  And how are tests possibly passing with this code?

Reply:
Please make sure to point me to the tests that would have failed due to this bug in the updates.